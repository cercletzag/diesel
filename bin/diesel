#!/bin/sh
cd diesel_cli && cargo run -- $@
