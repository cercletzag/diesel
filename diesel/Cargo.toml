[package]
name = "diesel"
version = "0.4.1"
authors = ["Sean Griffin <sean@seantheprogrammer.com>"]
license = "MIT OR Apache-2.0"
description = "A safe, extensible ORM and Query builder"
documentation = "http://sgrif.github.io/diesel/diesel/index.html"
homepage = "https://github.com/sgrif/diesel"
repository = "https://github.com/sgrif/diesel"
keywords = ["orm", "database", "postgres", "postgresql", "sql"]

[dependencies]
libc = "0.2.*"
pq-sys = { version = "^0.2.0", optional = true }
libsqlite3-sys = { version = "^0.4.0", optional = true }
byteorder = "0.3.*"
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git", optional = true }
chrono = { version = "^0.2.17", optional = true }

[dev-dependencies]
quickcheck = { git = "https://github.com/BurntSushi/quickcheck.git" }
dotenv = "^0.6.0"
tempdir = "^0.3.4"

[features]
default = ["postgres"]
unstable = []
large-tables = []
postgres = ["pq-sys"]
sqlite = ["libsqlite3-sys"]
