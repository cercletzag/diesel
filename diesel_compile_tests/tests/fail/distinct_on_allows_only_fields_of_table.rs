extern crate diesel;

use diesel::*;

table! {
    users {
        id -> Integer,
        name -> VarChar,
    }
}

table! {
    posts {
        id -> Integer,
        name -> Text,
        content -> Text,
    }
}

fn main() {
    let connection = PgConnection::establish("postgres://foo").unwrap();

    users::table.distinct_on(posts::id).get_results(&connection);

    posts::table.distinct_on((posts::name, users::name)).get_result(&connection);
}
