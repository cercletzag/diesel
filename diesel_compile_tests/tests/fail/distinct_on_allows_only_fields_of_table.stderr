error[E0277]: the trait bound `posts::columns::id: SelectableExpression<users::table>` is not satisfied
  --> tests/fail/distinct_on_allows_only_fields_of_table.rs:23:30
   |
23 |     users::table.distinct_on(posts::id).get_results(&mut connection);
   |                              ^^^^^^^^^ the trait `SelectableExpression<users::table>` is not implemented for `posts::columns::id`
   |
   = help: the following implementations were found:
             <posts::columns::id as SelectableExpression<JoinOn<Join, On>>>
             <posts::columns::id as SelectableExpression<Only<posts::table>>>
             <posts::columns::id as SelectableExpression<SelectStatement<FromClause<From>>>>
             <posts::columns::id as SelectableExpression<diesel::internal::table_macro::Join<Left, Right, Inner>>>
           and 2 others
   = note: required because of the requirements on the impl of `DistinctOnDsl<posts::columns::id>` for `users::table`

error[E0277]: the trait bound `(diesel::sql_types::Text, diesel::sql_types::Text): SingleValue` is not satisfied
  --> tests/fail/distinct_on_allows_only_fields_of_table.rs:25:18
   |
25 |     posts::table.distinct_on((posts::name, users::name)).get_result(&mut connection);
   |                  ^^^^^^^^^^^ the trait `SingleValue` is not implemented for `(diesel::sql_types::Text, diesel::sql_types::Text)`
   |
   = note: required because of the requirements on the impl of `DistinctOnDsl<(posts::columns::name, users::columns::name)>` for `posts::table`

error[E0277]: the trait bound `users::columns::name: SelectableExpression<posts::table>` is not satisfied
  --> tests/fail/distinct_on_allows_only_fields_of_table.rs:25:30
   |
25 |     posts::table.distinct_on((posts::name, users::name)).get_result(&mut connection);
   |                              ^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `SelectableExpression<posts::table>` is not implemented for `users::columns::name`
   |
   = help: the following implementations were found:
             <users::columns::name as SelectableExpression<JoinOn<Join, On>>>
             <users::columns::name as SelectableExpression<Only<users::table>>>
             <users::columns::name as SelectableExpression<SelectStatement<FromClause<From>>>>
             <users::columns::name as SelectableExpression<diesel::internal::table_macro::Join<Left, Right, Inner>>>
           and 2 others
   = note: required because of the requirements on the impl of `SelectableExpression<posts::table>` for `(posts::columns::name, users::columns::name)`
   = note: required because of the requirements on the impl of `DistinctOnDsl<(posts::columns::name, users::columns::name)>` for `posts::table`

error[E0271]: type mismatch resolving `<posts::table as AppearsInFromClause<users::table>>::Count == diesel::query_source::Once`
  --> tests/fail/distinct_on_allows_only_fields_of_table.rs:25:18
   |
25 |     posts::table.distinct_on((posts::name, users::name)).get_result(&mut connection);
   |                  ^^^^^^^^^^^ expected struct `diesel::query_source::Never`, found struct `diesel::query_source::Once`
   |
   = note: required because of the requirements on the impl of `AppearsOnTable<posts::table>` for `users::columns::name`
   = note: 1 redundant requirements hidden
   = note: required because of the requirements on the impl of `AppearsOnTable<posts::table>` for `(posts::columns::name, users::columns::name)`
   = note: required because of the requirements on the impl of `SelectableExpression<posts::table>` for `(posts::columns::name, users::columns::name)`
   = note: required because of the requirements on the impl of `DistinctOnDsl<(posts::columns::name, users::columns::name)>` for `posts::table`

error[E0277]: the trait bound `posts::table: TableNotEqual<users::table>` is not satisfied
  --> tests/fail/distinct_on_allows_only_fields_of_table.rs:25:18
   |
25 |     posts::table.distinct_on((posts::name, users::name)).get_result(&mut connection);
   |                  ^^^^^^^^^^^ the trait `TableNotEqual<users::table>` is not implemented for `posts::table`
   |
   = note: required because of the requirements on the impl of `AppearsInFromClause<users::table>` for `posts::table`
   = note: required because of the requirements on the impl of `AppearsOnTable<posts::table>` for `users::columns::name`
   = note: 1 redundant requirements hidden
   = note: required because of the requirements on the impl of `AppearsOnTable<posts::table>` for `(posts::columns::name, users::columns::name)`
   = note: required because of the requirements on the impl of `SelectableExpression<posts::table>` for `(posts::columns::name, users::columns::name)`
   = note: required because of the requirements on the impl of `DistinctOnDsl<(posts::columns::name, users::columns::name)>` for `posts::table`
